### PR TITLE
fstab merging to development-mk

### PR DESCRIPTION
**Pull request by erichendrickson**

### CODE REVIEW

**Comment by konstruktoid** [comment]
What's the purpose of this MR?

File: config/tmp.mount [hunk 1]
```diff
@@ -9,4 +9,4 @@ Before=local-fs.target
 What=tmpfs
 Where=/tmp
 Type=tmpfs
-Options=mode=1777,strictatime,nosuid,nodev
+Options=mode=1777,nodev,nosuid,noexec,relatime
```
Comment (konstruktoid) [inline config/tmp.mount]: This will break a standard installation, `noexec` tmp will force constant remounting.

File: scripts/08_fstab [hunk 2]
```diff
@@ -1,15 +1,93 @@
...
   echo "[$i] /etc/fstab, system/tmp.mount and system/var-tmp.mount"
 
-  cp ./config/tmp.mount /etc/systemd/system/tmp.mount
+  cp /etc/fstab /etc/fstab.orig
 
-  sed -i '/floppy/d' /etc/fstab
+  read -r -a devices <<< "$(echo $DEVICES | sed 's/\/dev\/s/\/dev\/xv/g')"
+
+  if [ "4" != "${#devices[@]}" ]; then
+    "You cannot have more or less than four devices. Exiting."
```
Comment (konstruktoid) [inline scripts/08_fstab]: Why?

File: scripts/08_fstab [hunk 3]
```diff
@@ -1,15 +1,93 @@
...
+    echo
+    exit 1
+  fi
+
+  if ! lsblk | grep /var; then
+    systemctl rsyslog stop
+
+    mkfs -t ext4 ${devices[0]}
```
Comment (konstruktoid) [inline scripts/08_fstab]: The whole `lsblk` is messy, and automagical partitioning is better left to the user imo.
And why `ext4`?

File: scripts/08_fstab [hunk 4]
```diff
@@ -1,15 +1,93 @@
...
+    mkfs -t ext4 ${devices[3]}
+    mkdir /mnt/newhome
+    mount ${devices[3]} /mnt/newhome
+    cp -apx /home/* /mnt/newhome
+    umount ${devices[3]}
+    mv /home /home.old
+    mkdir /home
+    echo "${devices[3]}	/home	ext4	rw,nodev,relatime,data=ordered	0 0" >> /etc/fstab
+    mount -a
+    cd ~/MK-CIS-hardening
```
Comment (konstruktoid) [inline scripts/08_fstab]: What's this?

File: scripts/08_fstab [hunk 5]
```diff
@@ -1,15 +1,93 @@
...
 
-  if ! grep -i '/proc' /etc/fstab 2>/dev/null 1>&2; then
-    echo 'none /proc proc rw,nosuid,nodev,noexec,relatime,hidepid=2 0 0' >> /etc/fstab
+  rm -rf /mnt/newvar
+  rm -rf /mnt/newvarlog
+  rm -rf /mnt/newvarlogaudit
+  rm -rf /mnt/newhome
+
+  if [[ "$(mount | grep /dev/shm)" != *"tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime)"* ]]; then
+    echo "tmpfs	/dev/shm	tmpfs	rw,nosuid,nodev,noexec,relatime	0 0" >> /etc/fstab
```
Comment (konstruktoid) [inline scripts/08_fstab]: Why going back to fstab when using systemd?

File: tests/04_disablemnt.bats [hunk 6]
```diff
@@ -29,7 +29,8 @@ load test_helper
 
 @test "Verify that kernel module squashfs is disabled" {
   run bash -c "modprobe -n -v squashfs | grep 'install /bin/true'"
-  [ "$status" -eq 0 ]
+  run bash -c "modprobe -c | grep squashfs | grep 'install /bin/true'"
+  [ "$?" -eq 0 ] || [ "$status" -eq 0 ]
```
Comment (konstruktoid) [inline tests/04_disablemnt.bats]: Why? `[ "$status" -eq 0 ]` is sufficent as a `bats` result.

File: tests/08_fstab.bats [hunk 7]
```diff
@@ -56,7 +56,7 @@ load test_helper
   [ "$status" -eq 0 ]
 }
 
-@test "Verify that /home is a seperate partition" {
+@test "Verify that /home is a separate partition" {
```
Comment (konstruktoid) [inline tests/08_fstab.bats]: Thanks, will fix the spelling.

File: ubuntu.sh [hunk 8]
```diff
@@ -20,50 +20,55 @@ for s in ./scripts/[0-9_]*; do
   source "$s"
 done
 
+# New ordering
```
Comment (konstruktoid) [inline ubuntu.sh]: A new order and you only enable three functions?
```# New ordering
f_pre
f_disablemnt
f_fstab```

**Comment by erichendrickson** [comment]
Very sorry. I was making a version for AWS on a forked repo. I supposed I merged to your repo by mistake. You can close this branch if you wish.

**Comment by erichendrickson** [comment]
Delete the branch, I mean. Once again, very sorry.

**Comment by konstruktoid** [comment]
No worries @erichendrickson, have a great day.

**Pull request closed without merging**